.github/actions/learning-path-staleness-check/index.js

```diff
@@ -0,0 +1,288 @@
+  core.setOutput('modifiedFilesToCommit', modifiedFilesToCommit.join(' '))
+}
+
+function ReplaceOldWithNewText(content, oldText, newText)
+{
+  return content.replaceAll(oldText, newText);
+}
+
+function UpdateModifiedFiles(fileName, path, learningPathFile)
+{
+  modifiedFilesUrlToFileName[path] = fileName;
+
+  modifiedFilesDict[path] = modifiedFilesDict[path] ? modifiedFilesDict[path] : new Set();;
+  modifiedFilesDict[path].add(learningPathFile);
+
+  modifiedFiles = new Set();
+  for (currPath in modifiedFilesDict)
+  {
+    const fileName = modifiedFilesUrlToFileName[currPath];
+    modifiedFiles.add(AssembleModifiedFilesOutput(fileName, currPath, Array.from(modifiedFilesDict[currPath])));
+  }
+}
+
+function AssembleModifiedFilesOutput(fileName, path, learningPathFiles)
+{
+  return CreateLink(fileName, path, undefined) + separator + BoldedText(learningPathFiles.join(" "));
+}
+
+function BoldedText(text)
+{
+  return "**" + text + "**";
+}
+
+function UpdateManuallyReview(fileName, path, learningPathFile, learningPathLineNumber, lineNumber = undefined)
+{
+  manuallyReview.add(AssembleOutput(fileName, path, undefined, lineNumber, undefined, learningPathFile, learningPathLineNumber))
+}
+
+function UpdateOutOfSync(link, learningPathFile)
+{
+  outOfSync.add(link + separator + BoldedText(learningPathFile))
+}
+
+function UpdateSuggestions(fileName, oldPath, newPath, learningPathFile, learningPathLineNumber, oldLineNumber, newLineNumber)
+{
+  suggestions.add(AssembleOutput(fileName, oldPath, newPath, oldLineNumber, newLineNumber, learningPathFile, learningPathLineNumber))
+}
+
+function CreateLink(fileName, path, lineNumber)
+{
+  var codeFileLink = "[" + fileName + "]" + "(" + path + ")"
+  return AppendLineNumber(codeFileLink, lineNumber)
+}
+
+function AssembleOutput(fileName, oldPath, newPath, oldLineNumber, newLineNumber, learningPathFile, learningPathLineNumber)
+{
+  var codeFileLink = CreateLink(fileName, oldPath, oldLineNumber)
+
+  if (newPath && newLineNumber) {
+    codeFileLink += oldNewLinkSeparator + CreateLink(fileName, newPath, newLineNumber)
+  }
+
+  return codeFileLink + separator + BoldedText(AppendLineNumber(learningPathFile, learningPathLineNumber, undefined));
+}
+
+function AppendLineNumber(text, lineNumber)
+{
+  if (!lineNumber) { return text }
+
+  return text + " " + linePrefix + lineNumber
+}
+
+function CheckForEndOfLink(str, startIndex)
+{
+  const illegalCharIndex = str.substr(startIndex).search("[(), '\`\"\}\{]|\. "); // This regex isn't perfect, but should cover most cases.
+  return illegalCharIndex;
+}
+
+function StripLineNumber(link, linePrefixIndex)
+{
+  return link.substring(0, linePrefixIndex);
+}
+
+function GetContent(path) {
+  try {
+    return actionUtils.readFileSync(path)
+  }
+  catch (error) {}
+
+  return undefined;
+}
+
+function ConstructOutputText(core)
+{
+  var manuallyReviewSection = "<h2>Manually Review:</h2>" + Array.from(manuallyReview).join("<br />") + "<br />";
```

Comment:
nit: How it's written is a far bit harder to follow (and more prone to easy-to-miss bugs in variable usage)  than building the PR body as we go,


e.g.
```javascript
var body =""
if (manuallyReview.size) {
  body += "<h2>Manually Review:</h2>" + Array.from(manuallyReview).join("<br />") + "<br />"
}

...


```